### DISCUSSION TITLE
jvm/bytecode

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#278] [Announcements]
# jvm/bytecode

本文主要介绍javap与字节码的关系，以及如何通过javap命令来查看字节码。

https://javabetter.cn/jvm/bytecode.html

**Comment by nanfangqiulin** [comment]
大牛，本文常量池这一节，只讲了两个常量关联的，似乎还有几个常量没说？比如
  #9 = Utf8               Code
  #10 = Utf8               LineNumberTable
  #11 = Utf8               LocalVariableTable
  #12 = Utf8               this

**Comment by chaofengdev** [comment]
我可以将javap输出的结果理解为，是对上面的xxx.class的十六进制的翻译结果吗？
本质都是xxx.class文件的不同表示形式？

**Reply by chaofengdev** [reply]
查了一下：本质上，.class 文件是以二进制格式存储的，而 javap 工具是将这些信息提取出来并以人类可读的形式展示出来。.class 文件中的数据经过了高度组织，javap 提供的输出是对这些数据的一种解读。

**Comment by qqluo666** [comment]
super_class: #4 // java/lang/Object 这个父类的索引指向的应该是常量池里下标为4的常量吧
